src/z3c/relationfield/relation.py

```diff
@@ -73,10 +79,18 @@ def __eq__(self, other):
     def __ne__(self, other):
         return not self.__eq__(other)
 
-    def __cmp__(self, other):
-        if other is None:
-            return cmp(self._sort_key(), None)
-        return cmp(self._sort_key(), other._sort_key())
+    def __lt__(self, other):
+        ''' Relations are sorted by default on a combination of the relation name,
+        the path of the object the relation is one and the path of the object
+        the relation is pointing to.
+        '''
+        if (self.from_attribute or '') < (other.from_attribute or ''):
+            return True
+        if (self.from_path or '') < (other.from_path or ''):
+            return True
+        if (self.to_path or '') < (other.to_path or ''):
+            return True
+        return False
```

Comment:
Could this be simplified to
```python
return self._sort_key() < other._sort_key()
```
?

And also maybe we should check object types, i.e.

```python
if not isinstance(other, RelationValue):
    return NotImplemented
```

Reply:
Checking the type is a nice plus.
comparing `obj._sort_key()` will return errors when the tuple contains `None`.

Reply:
Oh, there's an `__eq__`?

Do we preserve the requirement that hash(x) == hash(y) when `x.__eq__(y)` returns True?  If not, that's bad.